lib/screens/forgot_password/forgot_password.dart

```diff
@@ -21,24 +21,36 @@ class ForgotPasswordScreen extends StatelessWidget {
             Scaffold.of(context).showSnackBar(
               SnackBar(content: Text(state.error)),
             );
-          } else if (state is ForgotWrongOTP) {
+          } else if (state is ForgotOTPFailure) {
             Scaffold.of(context).showSnackBar(
               SnackBar(content: Text("Wrong OTP")),
             );
+          } else if (state is ForgotEmailFailure) {
+            Scaffold.of(context).showSnackBar(
+              SnackBar(content: Text("Please check email")),
+            );
+          } else if (state is ForgotResetFailure) {
+            Scaffold.of(context).showSnackBar(
+              SnackBar(content: Text("Failed to reset password")),
+            );
           }
         },
         builder: (context, state) {
-          if (state is ForgotInitial) {
+          if (state is ForgotEmailInitial) {
             return _initialForgotPassword(context);
           } else if (state is ForgotLoading) {
             return _buildLoading();
-          } else if (state is ForgotEnterOTP) {
+          } else if (state is ForgotEmailFailure) {
+            return _initialForgotPassword(context);
+          } else if (state is ForgotOTPInitial) {
             return _enterOTP(context);
-          } else if (state is ForgotWrongOTP) {
+          } else if (state is ForgotOTPFailure) {
             return _enterOTP(context);
-          } else if (state is ForgotCreateNewPassword) {
+          } else if (state is ForgotResetInitial) {
+            return _resetPassword(context);
+          } else if (state is ForgotResetFailure) {
             return _resetPassword(context);
-          } else if (state is ForgotPasswordSuccess) {
+          } else if (state is ForgotResetSuccess) {
             return _passwordResetSuccess();
           } else {
```

Comment:
There is a problem here. consider this scenario: 
User enters their email, moves to the otp screen, the internet goes off, and they try to check their otp. A network error occurs, the code runs this else function and the user goes to the enter email screen. We'll have a call over how you can solve this issue.

Reply:
Okay. Let's discuss this over a call.